docs/Binding-and-project-context-evaluation.md

```diff
@@ -0,0 +1,73 @@
+# Binding and project context evaluation
+
+## Overview
+
+Since .NET SDK 7.0.100 template engine supports binding of symbols to various external sources, including MSBuild properties of the current project.
+The feature is available for both `dotnet new` and Visual Studio.
+
+## `bind` symbols
+
+The symbol binds value from external sources. 
+By default, the following sources are available:
+- host parameters - parameters defined at certain host. For .NET SDK the following parameters are defined: `HostIdentifier: dotnetcli`, `GlobalJsonExists: true/false`.  Binding syntax is `host:<param name>`, example: `host:HostIdentifier`. 
+- environment variables - allows to bind environment allowed. Binding syntax is `env:<environment variable name>`, example: `env:MYENVVAR`. 
+
+It is also possible to bind the parameter without the prefix as the fallback behavior: `HostIdentifier`, `MYENVVAR`.
```

Comment:
It might be worth to mention the order of sources search in this case (seems to be envirenment, then host - https://github.com/dotnet/templating/blob/main/src/Microsoft.TemplateEngine.Edge/Components.cs#L35-L36)

Reply:
It's actually based on component priority. I see that mentioned in the 'inside template engine api' doc, but I still feel the default behavior (environment has priority 0) should be metnioned here as well.